library/Vanilla/Utility/ContainerUtils.php

```diff
@@ -67,9 +67,39 @@ public static function currentTheme(): ReferenceInterface {
     public static function cacheBuster(): ReferenceInterface {
         return new Callback(
             function (ContainerInterface $dic) {
-                $cacheBuster = $dic->get(\Vanilla\Web\Asset\DeploymentCacheBuster::class);
+                $cacheBuster = $dic->get(DeploymentCacheBuster::class);
                 return $cacheBuster->value();
             }
         );
     }
+
+    /**
+     * Replace one type of object in the container with another type of object. Existing shared instances will be
+     * overwritten. Optionally, an alias may be created from the original type to the new type.
+     *
+     * Sometimes an object has limitations or shortcomings that could be resolved by something like a decorator, where
+     * a drop-in replacement wraps existing functionality in enhancements or customizations. This method could configure
+     * a container to use the decorator, replacing stored instances and ensuring new requests to the container would
+     * receive the decorator instead of the original class.
+     *
+     * @param Container $container Container to configure.
+     * @param string $target Container rule to target for replacement. Shared instances will be overwritten.
+     * @param string $replacement Container rule used to determine what will be replace the target in the container.
+     * @param bool $addAlias Should an alias from the target rule be made to the replacement?
+     */
+    public static function replace(
+        Container $container,
+        string $target,
+        string $replacement,
+        bool $addAlias = false
```

Comment:
I would say this should default to true or not be a parameter at all. Otherwise, if there isn't really a replacement happening and the method has inconsistent behavior depending on whether or not the container has the instance.

IMO you think about the following:

1. Go for it and get rid of `$addAlias` and just make this the behavior.
2. Just remove the instance if it is there and add a rule. Let the container do the creation when it is called. This one may be a bit controversial since we've had problem with the log in particular getting injected and called before it is instantiated. On the other hand, the caller could always do the instantiation themself. If they don't want to instantiate then they are left with having to write the code themselves.

Reply:
> Go for it and get rid of $addAlias and just make this the behavior.

I was waffling on this. There might be a scenario where you'd only want to replace the current instance, but we don't have that need now and I can't think of a concrete scenario where it'd be warranted. I'm A-OK with removing the parameter.

> Just remove the instance if it is there and add a rule. Let the container do the creation when it is called. This one may be a bit controversial since we've had problem with the log in particular getting injected and called before it is instantiated. On the other hand, the caller could always do the instantiation themself. If they don't want to instantiate then they are left with having to write the code themselves.

If there's already an instance, I like the idea of having a little more control over when the replacement will be instantiated. But again, our immediate need doesn't require this and I'm okay with revisiting it in the future, as necessary.